### PR TITLE
add spherical geometry support in compressible

### PR DESCRIPTION
**Pull request by zhichen3**
This pr addresses issue #203

### CODE REVIEW

**Comment by zhichen3** [comment]
still need to update how artificial viscosity is changed due to spherical geometry, there is a div u involved in calculating viscosity coefficient.

**Comment by zhichen3** [comment]
also need to address BC correctly.

**Comment by zingale** [comment]
can you split this up into a few different PRs?

For example:
1. a PR with the Riemann interface changes
2. a PR with the change in the artifical viscosity for Cartesian
3. the spherical stuff

some of the Riemann changes are similar to what I've been doing to allow us to have more solvers.

File: pyro/compressible/simulation.py [hunk 1]
```diff
@@ -193,33 +207,105 @@ def evolve(self):
...
         for n in range(self.ivars.nvar):
             var = self.cc_data.get_var_by_index(n)
 
-            var.v()[:, :] += \
-                dtdx*(Flux_x.v(n=n) - Flux_x.ip(1, n=n)) + \
-                dtdy*(Flux_y.v(n=n) - Flux_y.jp(1, n=n))
+            var.v()[:, :] += dtdV * \
+                (F_x.v(n=n)*myg.Ax.v() - F_x.ip(1, n=n)*myg.Ax.ip(1) +
+                 F_y.v(n=n)*myg.Ay.v() - F_y.jp(1, n=n)*myg.Ay.jp(1))
+
```
Comment (zingale) [inline pyro/compressible/simulation.py:288]: I don't think that this is right.  Only the divergence terms should have the area factors.  For non-Cartesian, the grad{p} term is different and we need to address that separately.

Comment (zhichen3) [reply pyro/compressible/simulation.py:288]: gradp is added after this , where I apply external source. For non cartesian, xmom and ymom flux don't have pressure, you can check consFlux.

Comment (zingale) [reply pyro/compressible/simulation.py:288]: ah okay.  Can you update the comment then to say that it also adds grad p for the coord_type = 1 case?

Comment (zhichen3) [reply pyro/compressible/simulation.py:290]: added more comments.

**Comment by zingale** [comment]
okay, I think this is good.  There might be some changes in the future, but let's merge this and then we can play with it more.

**Pull request merged**